filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilder.scala

```diff
@@ -55,10 +55,12 @@ object BlobPathBuilder {
   }
 }
 
-class BlobPathBuilder(credential: AzureSasCredential, container: String, endpoint: String) extends PathBuilder {
+class BlobPathBuilder(blobTokenGenerator: BlobTokenGenerator, container: String, endpoint: String) extends PathBuilder {
 
+  val credential: AzureSasCredential = new AzureSasCredential(blobTokenGenerator.getAccessToken)
```

Comment:
Thinking Out Loud: I wonder if we should be calling `getAccessToken` from `[Blob]PathBuilder.build` so that it (theoretically) happens closer to object access. This might be moot once we figure out how to handle token refresh... or maybe changing when we call `getAccessToken` would make it easier to reason about and solve token refresh.